crates/sui-core/src/state_accumulator.rs

```diff
@@ -118,33 +119,44 @@ impl StateAccumulator {
             .flat_map(|fx| {
                 fx.unwrapped()
                     .iter()
-                    .map(|(oref, _owner)| (oref.0, oref.1))
-                    .collect::<Vec<(ObjectID, SequenceNumber)>>()
+                    .map(|(oref, _owner)| (*fx.transaction_digest(), oref.0, oref.1))
+                    .collect::<Vec<(TransactionDigest, ObjectID, SequenceNumber)>>()
```

Comment:
i know it was like this already but do you actually need the inner collect()s here? flat_map should work fine with iterators